### DISCUSSION TITLE
Frontera: Encounter error when running SeisSol after successful compiling it

### DISCUSSION THREAD

**Discussion by jwjeremy** [SeisSol/SeisSol#1154] [Q&A] [answered]
**Describe the bug**
I followed the[ seissol Frontera docs](https://seissol.readthedocs.io/en/latest/frontera.html#frontera) in compiling a new spack seissol-env environment and successfully compiled the SeisSol. However, I encounter an error when running the seissol at the stage of Reading PUML mesh and Inferring boundary format with the following HDF5 error: 
```
Wed Aug 28 17:33:59, Info:  Reading PUML mesh 
Wed Aug 28 17:33:59, Info:  Inferring boundary format. 
HDF5-DIAG: Error detected in HDF5 (1.12.2) MPI-process 1:
  #000: /tmp/jwjeremy/spack-stage/spack-stage-hdf5-1.12.2-jfiibraggrkahq5xlsdkuqpyu2hufiq7/spack-src/src/H5F.c line 620 in H5Fopen(): unable to open file
    major: File accessibility
    minor: Unable to open file
```
I also could not find the file listed in the error message. 

**Expected behavior**
Run SeisSol simulation

**To Reproduce**
Frontera is used. Module list:
```
Currently Loaded Modules:
  1) intel/19.1.1                                 18) impalajit/main-intel-19.1.1.217-omg4mnr
  2) impi/19.0.9                                  19) curl/7.29.0-intel-19.1.1.217-kiwu5vb
  3) git/2.24.1                                   20) ncurses/6.4-intel-19.1.1.217-kibbtxs
  4) autotools/1.2                                21) readline/8.2-intel-19.1.1.217-j7gf2vu
  5) cmake/3.24.2                                 22) unzip/6.0-intel-19.1.1.217-s2gzz22
  6) hwloc/1.11.12                                23) lua/5.3.2-intel-19.1.1.217-lnzubir
  7) xalt/2.10.34                                 24) yaml-cpp/0.6.2-intel-19.1.1.217-p36vrxa
  8) TACC                                         25) easi/1.2.0-intel-19.1.1.217-edmp22l
  9) python3/3.9.2                                26) eigen/3.4.0-intel-19.1.1.217-wni5u35
 10) intel-mpi/2019.0.9-intel-19.1.1.217-plmutsc  27) libxsmm/1.17-intel-19.1.1.217-7lkvdiy
 11) pkgconf/1.9.5-intel-19.1.1.217-tnsoy7m       28) memkind/1.13.0-intel-19.1.1.217-sdo3wzs
 12) zlib-ng/2.1.4-intel-19.1.1.217-xwszawp       29) metis/5.1.0-intel-19.1.1.217-yjhymv7
 13) hdf5/1.12.2-intel-19.1.1.217-jfiibra         30) parmetis/4.0.3-intel-19.1.1.217-q4zdww5
 14) netcdf-c/4.7.4-intel-19.1.1.217-id5g7mz      31) python/3.9.2-intel-19.1.1.217-ixn2w76
 15) numactl/2.0.14-intel-19.1.1.217-sqw4qpm      32) py-pspamm/develop-intel-19.1.1.217-dh5gy3d
 16) asagi/1.0.1-intel-19.1.1.217-b5l4bhl         33) seissol-env/develop-intel-19.1.1.217-bqiq6yg
 17) cxxtest/develop-intel-19.1.1.217-yavleqi
```

**Additional context**
I also try using the pre-compiled seissol-env in Frontera with the following commands to updates the modules 
`module use /work2/09160/ulrich/frontera/spack/share/spack/modules/linux-centos7-cascadelake`
`module load seissol-env`

And the SeisSol simulation encounters the same error during reading the PUML file with HDF5 error as the following.
 ```
  #000: /tmp/ulrich/spack-stage/spack-stage-hdf5-1.12.2-sji7bdxhcoqdeyaxr67uvxuocveqa34e/spack-src/src/H5F.c line 620 in H5Fopen(): unable to open file
    major: File accessibility
    minor: Unable to open file
```

Thanksss!

**Comment by davschneller** [comment, marked as answer]
Hi, that would mean that it cannot find the mesh file. From where are you executing SeisSol, and where is your mesh file located?

(and yes, we may need a more descriptive error message here)

You can also bypass the inference by using `pumlboundaryformat='i32'` (unless you're using something else here)—but if the mesh file cannot be found, it should still get stuck.

... That makes me wonder if it would make sense to determine the mesh file path relative to the parameter file path (like we do for easi now), not to the execution directory?

**Reply by jwjeremy** [reply]
Thanks so much! @davschneller, I found a typo when scp the mesh file... 
I also need this option `pumlboundaryformat='i32'; otherwise, it would encounter the unknown boundary format error. I do use the relative to the parameter file path (which is the same as the execution directory)

**Reply by davschneller** [reply]
Great to hear!
Is it possible to get the text the error message was saying exactly there? (may be useful to avoid future errors in that area)

**Reply by jwjeremy** [reply]
Sure!! 

Error when not using the option `pumlboundaryformat='i32` in seissol parameter file section `&MeshNml /`

```
Wed Aug 28 17:52:10, Info:  Reading PUML mesh 
Wed Aug 28 17:52:10, Info:  Inferring boundary format. 
Wed Aug 28 17:52:10, Info:  Boundary format given in PUML file. 
Wed Aug 28 17:52:10, Error: Unkown boundary format given in PUML file: "i32" 
Backtrace:
SeisSol_Release_dskx_4_elastic(_ZN5utils6LoggerD1Ev+0x319) [0x478639]
SeisSol_Release_dskx_4_elastic() [0x51ddab]
SeisSol_Release_dskx_4_elastic() [0x51e84f]
SeisSol_Release_dskx_4_elastic() [0x517e16]
SeisSol_Release_dskx_4_elastic() [0x41bdd5]
/usr/lib64/libc.so.6(__libc_start_main+0xf5) [0x2ba073384555]
SeisSol_Release_dskx_4_elastic() [0x41b27f]
Aborted
```

**Reply by davschneller** [reply]
Thanks a lot!
That error is exactly what is not supposed to happen in such a situation; but I now know why it happens; thanks again! :)

**Reply by vikaskurapati** [reply]
> ... That makes me wonder if it would make sense to determine the mesh file path relative to the parameter file path (like we do for easi now), not to the execution directory?

I think that is better, too! It would be great if there is a possibility of giving the absolute path or every file relative to the parameter file path instead of the execution directory or the location of the SeisSol binary.
